### PR TITLE
Fix/proper deallocation

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- [x] fixes pg init in webserver and storage
- [x] fixes redis init in webserver and adds dependency in docker-compose 
- [x] moves common functionality to servicelib (packages/service-library/src/servicelib/aiopg_utils.py)
- [x] enhances log when when pg retry policy fails

In addition
 - upgrades tenacity
 - upgrades to postgres:10.11 (some post recommends to upgrade if many disconnects happen...)

 

## Checklist

- [x] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: packages/service-library/tests/with_postgres/test_aiopg_utils.py [hunk 1]
```diff
@@ -34,36 +41,116 @@
...
+        assert dec_go.total_retry_count() == PostgresRetryPolicyUponOperation.ATTEMPTS_COUNT+1
 
         # goes and keeps count of all retrials
-        await go(engine, gid=2)
-        assert go.total_retry_count() == PostgresRetryPolicyUponOperation.ATTEMPTS_COUNT+2
+        await dec_go(engine, gid=2)
+        assert dec_go.total_retry_count() == PostgresRetryPolicyUponOperation.ATTEMPTS_COUNT+2
 
 
 #@pytest.mark.skip(reason="UNDER DEVELOPMENT")
```
Comment (sanderegg) [inline packages/service-library/tests/with_postgres/test_aiopg_utils.py]: remove comment?

File: services/sidecar/requirements/_base.txt [hunk 2]
```diff
@@ -4,22 +4,17 @@
...
 docker==3.7.2
-idna-ssl==1.1.0           # via aiohttp
-idna==2.8                 # via idna-ssl, requests, yarl
+idna==2.8                 # via requests
 kombu==4.5.0
 minio==4.0.16
-multidict==4.5.2          # via aiohttp, yarl
 networkx==2.3
 pika==1.0.1
 psycopg2-binary==2.8.3
```
Comment (sanderegg) [inline services/sidecar/requirements/_base.txt]: could we update to 2.8.4? it's python 3.8 compatible.

File: services/storage/src/simcore_service_storage/db.py [hunk 3]
```diff
@@ -1,61 +1,66 @@
...
+        )
+
+    log.info("Creating pg engine for %s", dsn)
+    for attempt in Retrying(**PostgresRetryPolicyUponInitialization(log).kwargs):  # pylint: disable=not-an-iterable
+        with attempt:
+            engine = await create_pg_engine(dsn,
+                minsize=pg_cfg['minsize'],
+                maxsize=pg_cfg['maxsize']
+            )
+            await raise_if_not_responsive(engine)
```
Comment (sanderegg) [inline services/storage/src/simcore_service_storage/db.py]: cool!

File: services/web/server/requirements/_base.txt [hunk 4]
```diff
@@ -61,7 +61,7 @@ shortuuid==0.5.0          # via aio-pika
 six==1.12.0               # via aadict, asset, cryptography, isodate, jsonschema, openapi-core, openapi-spec-validator, pyrsistent, python-engineio, python-socketio, tenacity
```
Comment (sanderegg) [inline services/web/server/requirements/_base.txt]: update psychopg 2.8.4 aswell??

File: services/sidecar/requirements/_base.txt (hunk 2 above)
Comment (pcrespov) [reply services/sidecar/requirements/_base.txt]: done

**Pull request merged**